framework/source/class/qx/event/handler/GestureCore.js

```diff
@@ -682,27 +718,125 @@ qx.Bootstrap.define("qx.event.handler.GestureCore", {
+      domEvent.pointerType = "wheel";
+      this._fireEvent(domEvent, "roll", domEvent.target || target);
+    },
+
+    /**
+     * Triggers the adaptative roll scrolling.
+     *
+     * @param target {Element} event target
+     */
+    __performAdaptativeRollScrolling: function (target) {
+      var rollFactor = qx.event.handler.GestureCore.WHEEL_ROLL_FACTOR;
+      if (qx.util.Wheel.IS_TOUCHPAD) {
+        // The domEvent was generated by a touchpad
+        rollFactor = qx.event.handler.GestureCore.TOUCHPAD_ROLL_FACTOR;
+      }
+      this.__lastRollEventTime = new Date().getTime();
+      var reLength = this.__rollEvents.length;
+      for (var i = 0; i < reLength; i++) {
+        var domEvent = this.__rollEvents[i];
+        this.__fireRollEvent(domEvent, target, rollFactor);
+      }
+      this.__rollEvents = [];
+    },
+
+    /**
+     * Ends touch pad detection process.
+     */
+    __endTouchPadDetection: function () {
+      if (this.__rollEvents.length > qx.event.handler.GestureCore.TOUCHPAD_WHEEL_EVENTS_THRESHOLD) {
+        qx.util.Wheel.IS_TOUCHPAD = true;
+      } else {
+        qx.util.Wheel.IS_TOUCHPAD = false;
+      }
+      if (qx.core.Environment.get("qx.debug.touchpad.detection")) {
+        qx.log.Logger.debug(this, "IS_TOUCHPAD : " + qx.util.Wheel.IS_TOUCHPAD);
+      }
+      this.__touchPadDetectionPerformed = true;
+    },
+
+    /**
+     * Is touchpad detection enabled ? Default implementation activates it only for Mac OS after Sierra (>= 10.12).
+     * @return {boolean} true if touchpad detection should occur.
+     * @internal
+     */
+    _isTouchPadDetectionEnabled: function () {
+      return qx.core.Environment.get("os.name") == "osx" && qx.core.Environment.get("os.version") >= "10.12";
+    },
+
+    /**
+     * Fires a roll event after determining the roll factor to apply. Mac OS Sierra (10.12+)
+     * introduces a lot more wheel events fired from the trackpad, so the roll factor to be applied
+     * has to be reduced in order to make the scrolling less sensitive.
+     *
+     * @param domEvent {Event} DOM event
      * @param type {String} The type of the dom event
      * @param target {Element} event target
      */
     _fireRoll : function(domEvent, type, target) {
       if (domEvent.type === qx.core.Environment.get("event.mousewheel").type) {
-        domEvent.delta = {
-          x: qx.util.Wheel.getDelta(domEvent, "x") * qx.event.handler.GestureCore.ROLL_FACTOR,
-          y: qx.util.Wheel.getDelta(domEvent, "y") * qx.event.handler.GestureCore.ROLL_FACTOR
-        };
-        domEvent.delta.axis = Math.abs(domEvent.delta.x / domEvent.delta.y) < 1 ? "y" : "x";
-        domEvent.pointerType = "wheel";
+        if (this._isTouchPadDetectionEnabled()) {
+          var now = new Date().getTime();
+          if (this.__lastRollEventTime > 0 && now - this.__lastRollEventTime
+            > qx.event.handler.GestureCore.TOUCHPAD_WHEEL_EVENTS_TIMEOUT) {
```

Comment:
By the way, I prefer all variable declarations at the top of the functions in JavaScript since having them at the beginning of a block implies that there are block-level variables (declared with var) which JavavScript does not support. There are plenty of instances in qooxdoo of the style being used here, so this is not a mandated change, but I think it makes the code more clear.